### PR TITLE
AYS-160 | Registering with Invalid Domain Extension Email

### PR DESCRIPTION
**Pull request by MenekseYuncu**
Please provide a brief description of the changes made in this pull request.

### Checklist

Before submitting your pull request, ensure the following:

- [x] **Title and Branch Naming Conventions:**
    - The pull request title follows the
      standard: [Pull Request Naming Conventions](CONTRIBUTING.md#pull-request-naming-conventions).
    - The branch name follows one of the conventions outlined in
      the [Branch Naming Conventions](CONTRIBUTING.md#branch-naming-conventions).

- [x] **Local Testing:**
    - I have tested my changes locally on Postman, and they are working as expected.

- [x] **Code Quality:**
    - The code is formatted according to the project's coding guidelines and style.
    - The code has been reviewed to ensure its quality.
    - The code does not contain any issues flagged by SonarLint.

- [x] **Documentation:**
    - Necessary documentation has been added or existing documentation has been updated, specifically detailing changes
      made in Postman.

- [x] **Testing:**
    - Relevant unit tests have been written and included.
    - Relevant integration tests have been written and included.

- [x] **Reviewers and Assignees:**
    - Default reviewers have been assigned to this pull request.
    - Assignees have been added if necessary.

- [x] **Labels and Associations:**
    - No specific actions are required in the Labels and Associations section for this pull request.

- [x] **Jira Issue Transition:**
    - If this pull request is related to a Jira issue, ensure that the issue is moved to the "In Review" status on the
      Jira board.

### CODE REVIEW

**Review by moaydogdu — DISMISSED** [review]
LGTM!

File: src/main/java/org/ays/common/util/validation/EmailValidator.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+/**
+ * A custom validator implementation for the {@link Email} annotation.
+ * Validates whether the provided email invalid domain
+ * specified regular expression.
+ */
+
+class EmailValidator implements ConstraintValidator<Email, String> {
+
+    private static final String EMAIL_REGEX =
+            "^[a-zA-Z0-9]+([._-][a-zA-Z0-9]+)*@[a-zA-Z0-9]+" + "([.-]?[a-zA-Z0-9]+)+\\.[a-zA-Z]{2,}$";
```
Comment (agitrubard) [inline src/main/java/org/ays/common/util/validation/EmailValidator.java:16]: Burayla ilgili geliştirme yapacaktık, bu kısım tamamlandı mı?

File: src/main/java/org/ays/common/util/validation/EmailValidator.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+
+
+/**
+ * A custom validator implementation for the {@link Email} annotation.
+ * Validates whether the provided email invalid domain
+ * specified regular expression.
+ */
+
```
Comment (agitrubard) [inline src/main/java/org/ays/common/util/validation/EmailValidator.java:12]: Buradaki fazladan boşluğu silebiliriz

File: src/main/java/org/ays/common/util/validation/EmailValidator.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
+package org.ays.common.util.validation;
+
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+
+
```
Comment (agitrubard) [inline src/main/java/org/ays/common/util/validation/EmailValidator.java:6]: Buradaki fazladan boşluğu silebiliriz

File: src/main/java/org/ays/common/util/validation/Email.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Annotation to validate email using {@link EmailValidator}.
+ */
+
```
Comment (agitrubard) [inline src/main/java/org/ays/common/util/validation/Email.java:14]: Buradaki fazladan boşluğu silebiliriz

File: src/main/java/org/ays/common/util/validation/EmailValidator.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+     * @return true if the value is valid, false otherwise
+     */
+    @Override
+    public boolean isValid(String email, ConstraintValidatorContext constraintValidatorContext) {
+
+        if (!email.matches(EMAIL_REGEX)) {
+            return false;
+        }
+
+        return email.matches(EMAIL_REGEX);
```
Comment (agitrubard) [inline src/main/java/org/ays/common/util/validation/EmailValidator.java:44]: Aşağıdaki şekilde yazmamız yeterli olur. Zaten matches olmazsa false dönecek. Ek olarak burada email'in boş geldiği durumlarda regex ile karşılaştırmak yerine validasyondan geçmesini sağlamamız lazım çünkü öbür türlü `@NotBlank` anotasyonuyla çakışabilir.

```suggestion

        if (!String.hasText(email)) {
            return true;
        }

        return email.matches(EMAIL_REGEX);
```

**Review by agitrubard — CHANGES_REQUESTED** [review]

**Review by agitrubard — APPROVED** [review]

**Pull request merged**